components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -27,329 +41,271 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+    auto int_state_view  = int_state_field.get_view<mPack**>();
+    view_Nd<mMask,2> int_mask_view("",m_num_cols,m_num_levs); // Track mask of interpolated values
+    const view_Nd<mPack,2> ext_state_view(reinterpret_cast<mPack*>(ext_state_field.data()),
+                                          m_num_cols,m_num_src_levs);
+    // Masked values in the source data can lead to strange behavior in the vertical interpolation.
+    // We pre-process the data and map any masked values (sometimes called "filled" values) to the
+    // nearest un-masked value.
+    Real var_fill_value;
+    // WARNING!!! Assumes that all datafiles use the same FillValue.  It is unlikely that a user will use a mismatch of files
+    // with different defined FillValues, but if they do, this loop won't catch the change.
+    scorpio::get_variable_metadata(m_datafiles[0],name,"_FillValue",var_fill_value);
+    const int num_cols           = int_state_view.extent(0);
+    const int num_vert_packs     = int_state_view.extent(1);
+    const auto policy = ESU::get_default_team_policy(num_cols, num_vert_packs);
+    Kokkos::parallel_for("correct_for_masked_values", policy,
+       	       KOKKOS_LAMBDA(MemberType const& team) {
       const int icol = team.league_rank();
-      auto T_mid_1d = ekat::subview(T_mid,icol);
-      auto T_mid_ext_1d = ekat::subview(T_mid_ext,icol);
-      auto u_1d = ekat::subview(u,icol);
-      auto u_ext_1d = ekat::subview(u_ext,icol);
-      auto v_1d = ekat::subview(v,icol);
-      auto v_ext_1d = ekat::subview(v_ext,icol);
+      auto int_mask_view_1d  = ekat::subview(int_mask_view,icol);
+      auto int_state_view_1d = ekat::subview(int_state_view,icol);
+      Real fill_value;
+      int  fill_idx = -1;
+      // Scan top to surf and backfill all values near TOM that are masked.
+      for (int kk=0; kk<m_num_levs; ++kk) {
+        const auto ipack = kk / mPack::n;
+	const auto iidx  = kk % mPack::n;
+        // Check if this index is masked
+	if (!(int_state_view_1d(ipack)[iidx]==var_fill_value)) {
+	  fill_value = int_state_view_1d(ipack)[iidx];
+	  fill_idx = kk;
+	  for (int jj=0; jj<kk; ++jj) {
+            const auto jpack = jj / mPack::n;
+	    const auto jidx  = jj % mPack::n;
+	    int_state_view_1d(jpack)[jidx] = fill_value;
+	  }
+	  break;
+	}
+      }
+      // Now fill the rest, the fill_idx should be non-negative.  If it isn't that means
+      // we have a column that is fully masked 
+      for (int kk=fill_idx+1; kk<m_num_levs; ++kk) {
+        const auto ipack = kk / mPack::n;
+	const auto iidx  = kk % mPack::n;
+        // Check if this index is masked
+	if (!(int_state_view_1d(ipack)[iidx]==var_fill_value)) {
+	  fill_value = int_state_view_1d(ipack)[iidx];
+	} else {
+	  int_state_view_1d(ipack)[iidx] = fill_value;
+	}
+      }
+    });
+
+
+    // Vertical Interpolation onto atmosphere state pressure levels
+    if (m_src_pres_type == DYNAMIC) {
```

Comment:
Is the difference between STATIC and DYNAMIC that the former is a 1d pressure view (same for all cols), while the latter is a horiz varying field? If so, I think we should choose different names for the option. Static/dynamic make me think of something constant/varying _in time_, rather than space (and IIUC, in both these cases the pressure is changing in time).

Perhaps something like `UNIFORM` vs `HORIZ_VARYING`, or `VERTICAL_PROFILE` vs `3D_PROFILE`...or something conveying the meaning of 1d vs 3d.

Reply:
Great suggestion! Yeah, I'm not a fan of these names I chose, but didn't want to spin my wheels on names so I just moved on.

Re: your question.  DYANAMIC (as named now) would also involve time varying because we will be grabbing p_mid from the nudging data which could be time varying.  It might be better to define 3 modes: `SINGLE_COL_UNIFORM`, `HORIZ_VARYING` and `TIME_DEPENDENT`

Reply:
Maybe `SINGLE_VERTICAL_PROFILE` instead of `SINGLE_COL_UNIFORM`? I like the other two names.